### PR TITLE
Update dependencies, use rebar3

### PR DESCRIPTION
**Pull request by chiroptical**
I essentially ran `rebar3 update && rebar3 test` and it worked. So, here is a PR if you wanted to include it.

It is kind of weird to include `./rebar` in the project? We could use a tool, like `nix`, to pin down the version of `rebar3`?

I was asking about this on the erlang Slack and was curious how difficult it would be to update.

### CODE REVIEW

**Comment by chiroptical** [comment]
Also, I am happy to include CI if you would like to run tests on each PR.

File: Makefile [hunk 1]
```diff
@@ -3,27 +3,27 @@
...
-	@./rebar eunit
+	@rebar3 eunit
 
 clean:
-	@./rebar clean
+	@rebar3 clean
 
 neotoma.plt:
-	@dialyzer --build_plt --apps erts kernel stdlib compiler crypto hipe \
+	@rebar3 dialyzer -- --build_plt --apps erts kernel stdlib compiler crypto hipe \
```
Comment (chiroptical) [inline Makefile:15]: Pretty sure this is how you pass these arguments

Comment (chiroptical) [reply Makefile:15]: Also interested in trying out gradualizer if that is interesting to you.

**Review by seancribbs — APPROVED** [review]
Thanks!

**Pull request merged**